src/Main.elm

```diff
@@ -1,10 +1,197 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser
+import Data.Audience as Aud
+import Data.AudienceFolder as AudFolders
+import Debug exposing (toString)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick)
 
 
-main : Html msg
+main : Program () Model Msg
 main =
-    Html.text "There will be app soon!"
+    Browser.sandbox
+        { init = init
+        , update = update
+        , view = view
+        }
+
+
+type alias Model =
+    { folders : List AudFolders.AudienceFolder
+    , opened : Opened
+    , audience : List Aud.Audience
+    , breadcrumbs : Breadcrumbs
+    }
+
+
+type alias Opened =
+    { currentId : Maybe Int
+    , parentId : Maybe Int
+    , usedIdList : List Int
+    , parentName : List String
+    , state : Bool
```

Comment:
This is unnecessary. You already have this information in `currentId` It's either `Just a` so `state = True` or `Nothing`. This breaks rule "single source of truth"